frontend/src/components/pages/connect/CreateConnector.tsx

```diff
@@ -107,30 +120,43 @@ class CreateConnector extends PageComponent {
 
 interface ConnectorWizardProps {
     connectClusters: Array<ClusterConnectors>;
+    activeCluster: string;
 }
 
-function ConnectorWizard({ connectClusters }: ConnectorWizardProps) {
-    const history = useHistory();
-
+const ConnectorWizard = observer(({ connectClusters, activeCluster }: ConnectorWizardProps) => {
+    const history = appGlobal.history;
     const [currentStep, setCurrentStep] = useState(0);
-    const [activeCluster, setActiveCluster] = useState<string | null>(null);
-    const [selectedPlugin, setSelectedPlugin] = useState<ConnectorPlugin | null>(
-        null);
-    const [properties, setProperties] = useState('');
-    const [invalidValidationResult, setInvalidValidationResult] = useState<ConnectorValidationResult | null>(
-        null);
+    const [selectedPlugin, setSelectedPlugin] = useState<ConnectorPlugin | null>(null);
+    const [invalidValidationResult, setInvalidValidationResult] = useState<ConnectorValidationResult | null>(null);
     const [validationFailure, setValidationFailure] = useState<unknown>(null);
     const [creationFailure, setCreationFailure] = useState<unknown>(null);
     const [genericFailure, setGenericFailure] = useState<Error | null>(null);
+    const [stringifiedConfig, setStringifiedConfig] = useState<string>('');
+    const [parsedUpdatedConfig, setParsedUpdatedConfig] = useState<any | null>(null);
+    const [postCondition, setPostCondition] = useState<boolean>(false);
+    const [loading, setLoading] = useState<boolean>(false);
+    const [connectClusterStore, setConnectClusterStore] = useState(ConnectClusterStore.getInstance(activeCluster));
 
     useEffect(() => {
-        if (connectClusters.length) {
-            const clusterName = connectClusters[0].clusterName;
-            setActiveCluster(clusterName);
-            api.refreshClusterAdditionalInfo(clusterName, true);
-        }
-    }, [connectClusters]);
+        const init = async () => {
+            await connectClusterStore.setup();
+        };
+        init();
+    }, [connectClusterStore]);
 
+    useEffect(() => {
+        setConnectClusterStore(ConnectClusterStore.getInstance(activeCluster));
+    }, [activeCluster]);
+
+    useEffect(() => {
+        try {
+            setParsedUpdatedConfig(JSON.parse(stringifiedConfig));
+        } catch (e) {
+            setParsedUpdatedConfig(null);
+            setPostCondition(false);
+        }
+        setPostCondition(true);
+    }, [stringifiedConfig]);
```

Comment:
Those are a lot of "useState" for local state.
Seeing as the component is an observer, are there any downsides to putting all of that state into a `useLocalObservable`? 

If it works, we don't have to change it, just asking out of curiosity.

Reply:
I see we could use local Observable here and it would be more legible, I just kept adding `useState` because I didn't want to refactor too much, silly me